components/src/core/Drawer/DrawerNavGroup.tsx

```diff
@@ -49,17 +54,60 @@ const useStyles = makeStyles((theme: Theme) =>
             borderRadius: '0px 24px 24px 0px',
             opacity: 0.9,
         },
+        secondaryLevelListGroup: {
+            backgroundColor: theme.palette.type === 'light' ? white[200] : black['A200'],
+            paddingBottom: 0,
+            paddingTop: 0,
+        },
+        rightComponent: {
+            transitionDuration: '300ms',
+            cursor: 'inherit',
+            display: 'flex',
+            height: 48,
+            alignItems: 'center',
+            '&.expanded': {
+                transform: 'rotate(180deg)',
+            },
+        },
     })
 );
 
-export type NavItem = {
-    active?: boolean;
-    icon?: JSX.Element;
+export type NestedNavItem = {
+    // Icon on the right. Will be flipped upside down if item is expanded
+    rightComponent?: JSX.Element;
+
+    // onClick of the entire row
     onClick?: Function;
-    statusColor?: string;
-    subtitle?: string;
+
+    // onClick for the rightComponent
+    onClickIcon?: Function;
```

Comment:
Wouldn't they just put onClock on the item they pass in as rightComponent? I don't think this needs to be a prop.

Reply:
Because we are handling the expansion state for them.